di-ipv-evcs-stub/lambdas/src/handlers/evcsHandler.ts

```diff
@@ -0,0 +1,169 @@
+          decodeURIComponent(userId),
+          request,
+        );
+        break;
+      case "GET": {
+        let accessTokenVarified;
+        try {
+          accessTokenVarified = await verifyAccessToken(
+            validateAccessToken(event.headers?.Authorisation),
+          );
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          console.error(error);
+          return buildApiResponse({ errorMessage: error.message }, 400);
+        }
+
+        if (accessTokenVarified)
+          res = await processGetUserVCsRequest(decodeURIComponent(userId));
+        break;
+      }
+      default:
+        return buildApiResponse(
+          { errorMessage: "Not received correct request http method." },
+          400,
+        );
+    }
+
+    return buildApiResponse(res.response, res.statusCode);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    console.error(error);
+    return buildApiResponse({ errorMessage: error.message }, 500);
+  }
+}
+
+function parsePostRequest(event: APIGatewayProxyEvent): PostRequest {
+  console.info(`---Request parsing----`);
+  if (!event.body) {
+    throw new Error("Missing request body");
+  }
+
+  const postRequest = JSON.parse(event.body);
+  if (
+    !postRequest ||
+    !postRequest.persistVCs ||
+    postRequest.persistVCs.length <= 0 ||
+    !isValidCreateVcState(postRequest.persistVCs)
+  ) {
+    throw new Error("Invalid request");
+  }
+
+  return postRequest;
+}
+
+function parsePatchRequest(event: APIGatewayProxyEvent): PatchRequest {
+  console.info(`---Request parsing----`);
+  if (!event.body) {
+    throw new Error("Missing request body");
+  }
+
+  const patchRequest = JSON.parse(event.body);
+  if (
+    !patchRequest ||
+    !patchRequest.updateVCs ||
+    patchRequest.updateVCs.length <= 0 ||
+    !isValidUpdateVcState(patchRequest.updateVCs)
+  ) {
+    throw new Error("Invalid request");
+  }
+
+  return patchRequest;
+}
+
+function isValidCreateVcState(persistVCs: PersistVC[]): boolean {
+  for (const persistVC of persistVCs) {
+    if (!(persistVC.state in CreateVcStates)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function isValidUpdateVcState(updateVCs: UpdateVC[]): boolean {
+  for (const updateVC of updateVCs) {
+    if (!(updateVC.state in UpdateVcStates)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function validateAccessToken(authheader: string | undefined): string {
+  if (authheader === undefined) {
+    throw new Error("Request missing access token");
+  }
+  const parts = authheader!.split(" ");
+  if (parts.length != 2) {
+    throw new Error("Invalid access token value");
+  }
+  if (parts[0] != "Bearer") {
+    throw new Error("Access token type must be Bearer");
+  }
+  if (parts[1] === undefined || parts[1] === "") {
+    throw new Error("The access token value must not be null or empty string");
+  }
```

Comment:
Could this actually be null? We are technically only checking against undefined or empty string, so might be worth updating the error message to be accurate

Reply:
done